include/RAJA/exec-cuda/MemUtils_CUDA.hxx

```diff
@@ -62,37 +62,90 @@
 
 namespace RAJA
 {
+/// Maximum number of blocks that RAJA will launch
+/// threads are reused to finish all work
+#define RAJA_CUDA_MAX_NUM_BLOCKS (1024 * 16)
```

Comment:
Is the '16' here meant to be the same as the #define at line 78? If so, can we do this in such a way that they stay consistent (e.g., if the max reduction vars is change, then this will be changed automatically). Perhaps the best way to handle this is by making these const int variables?  Also, it would help to add some comments describing why these values were chosen.